tests/unit/pubsub_v1/subscriber/test_dispatcher.py

```diff
@@ -259,21 +259,22 @@ def test_modify_ack_deadline_splitting_large_payload():
 
     items = [
         # use realistic lengths for ACK IDs (max 176 bytes)
-        requests.ModAckRequest(ack_id=str(i).zfill(176), seconds=60)
+        requests.ModAckRequest(ack_id=str(i).zfill(176), seconds=60, future=None)
         for i in range(5001)
     ]
+    manager.send_unary_modack.return_value = (items, [])
     dispatcher_.modify_ack_deadline(items)
 
-    calls = manager.send.call_args_list
+    calls = manager.send_unary_modack.call_args_list
     assert len(calls) == 3
 
     all_ack_ids = {item.ack_id for item in items}
     sent_ack_ids = collections.Counter()
 
     for call in calls:
-        message = call.args[0]
-        assert message._pb.ByteSize() <= 524288  # server-side limit (2**19)
-        sent_ack_ids.update(message.modify_deadline_ack_ids)
+        modack_ackids = call[1]["modify_deadline_ack_ids"]
+        assert len(modack_ackids) <= dispatcher._ACK_IDS_BATCH_SIZE
+        sent_ack_ids.update(modack_ackids)
```

Comment:
This seems to have changed the nature of the test, is that a find/replace error?

We should be checking the message byte size here.

Reply:
True, merely summing the individual strings would be a bit off. But what if we actually build a `ModifyAckDeadlineRequest` instance in the test and check the size of that? Since that's the payload that is sent to the backend?

Reply:
I think the complexity of that isn't worth it.